src/client/utils/tracking/gtm.ts

```diff
@@ -0,0 +1,57 @@
+import { TypeOfContract } from 'data/graphql'
+import { OfferData } from 'pages/OfferNew/types'
+import { isBundle, isYouth } from 'pages/OfferNew/utils'
+import { captureSentryError } from 'utils/sentry-client'
+
+enum NoComboTypes {
+  NoCombo = 'NO_COMBO',
+  NoComboYouth = 'NO_COMBO_YOUTH',
+}
+
+type GAContractType = NoComboTypes | TypeOfContract
+
+interface GTMOfferData {
+  insurance_type: GAContractType
+  referral_code: 'yes' | 'no'
+  number_of_people: number
+  insurance_price: number
+}
+
+export interface DataLayerObject {
+  event: string
+  offerData?: GTMOfferData
+}
+
+const pushToGTMDataLayer = (obj: DataLayerObject) => {
+  const castedWindow = window as any
+  castedWindow.dataLayer = castedWindow.dataLayer || []
+  castedWindow.dataLayer.push(obj)
+}
+
+export const trackOfferGTM = (
+  offerData: OfferData,
+  referralCodeUsed: boolean,
+) => {
+  try {
+    const contractType = () => {
```

Comment:
Maybe we cold rename this to `getContractType` or `mapContractType` or something? Just to imply this variable is a funciton

Reply:
indeed!